Assets/Mirror/Editor/Weaver/Writers.cs

```diff
@@ -0,0 +1,237 @@
+using System;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+
+namespace Mirror.Weaver
+{
+
+    public class Writers
+    {
+        const int MaxRecursionCount = 128;
+
+        public static MethodReference GetWriteFunc(TypeReference variable, int recursionCount = 0)
+        {
+            if (recursionCount > MaxRecursionCount)
+            {
+                Weaver.Error("GetWriteFunc recursion depth exceeded for " + variable.Name + ". Check for self-referencing member variables.");
+                return null;
+            }
+
+            if (Weaver.WeaveLists.writeFuncs.ContainsKey(variable.FullName))
+            {
+                MethodReference foundFunc = Weaver.WeaveLists.writeFuncs[variable.FullName];
+                if (foundFunc.Parameters[0].ParameterType.IsArray == variable.IsArray)
+                {
+                    return foundFunc;
+                }
+            }
+
+            if (variable.IsByReference)
+            {
+                // error??
+                Weaver.Error("GetWriteFunc variable.IsByReference error.");
+                return null;
+            }
+
+            MethodDefinition newWriterFunc;
+
+            if (variable.IsArray)
+            {
+                TypeReference elementType = variable.GetElementType();
+                MethodReference elemenWriteFunc = GetWriteFunc(elementType, recursionCount + 1);
+                if (elemenWriteFunc == null)
+                {
+                    return null;
+                }
+                newWriterFunc = GenerateArrayWriteFunc(variable, elemenWriteFunc);
+            }
+            else
+            {
+                if (variable.Resolve().IsEnum)
+                {
+                    return Weaver.NetworkWriterWriteInt32;
+                }
+
+                newWriterFunc = GenerateWriterFunction(variable, recursionCount);
```

Comment:
Why is recursionCount not incremented here?

Reply:
This is a really poor name for that function,  It should be GenerateStructWriterFunction

the recursion happens when `GenerateStructWriterFunction` calls `GetWriterFunc`